features/flows/elastic/src/main/java/org/opennms/netmgt/flows/elastic/DocumentEnricher.java

```diff
@@ -221,19 +205,58 @@ private NodeInfoKey(final String location, final String ipAddress, final Context
         public boolean equals(Object o) {
             if (this == o) return true;
             if (o == null || getClass() != o.getClass()) return false;
-            final NodeInfoKey that = (NodeInfoKey) o;
-            return Objects.equals(location, that.location) &&
-                    Objects.equals(ipAddress, that.ipAddress) &&
-                    Objects.equals(contextKey, that.contextKey) &&
+            final NodeMetadataKey that = (NodeMetadataKey) o;
+            return Objects.equals(contextKey, that.contextKey) &&
                     Objects.equals(value, that.value);
         }
 
         @Override
         public int hashCode() {
-            return Objects.hash(location, ipAddress, contextKey, value);
+            return Objects.hash(contextKey, value);
         }
     }
 
+    private Optional<NodeDocument> getNodeInfoFromMetadataContext(ContextKey contextKey, String value) {
+        OnmsNode onmsNode = null;
+        if (contextKey != null && !Strings.isNullOrEmpty(value)) {
+            final List<OnmsNode> nodes = nodeDao.findNodeWithMetaData(contextKey.getContext(), contextKey.getKey(), value);
+
+            if (!nodes.isEmpty()) {
+                onmsNode = nodes.get(0);
```

Comment:
```suggestion
                return mapOnmsNodeToNodeDocument(nodes.get(0));
```

then can remove `onmsNode` local variable and other `if` statement bellow.